cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -353,8 +248,8 @@ private InetAddress getLocalHost() {
     return configKeys;
   }
 
-  private Map<String, Integer> getSystemServiceInstances() {
-    Map<String, Integer> instanceCountMap = new HashMap<String, Integer>();
+  private Map<String, Integer> getSystemServiceInstances(CConfiguration cConf) {
+    Map<String, Integer> instanceCountMap = new HashMap<>();
     for (Map.Entry<String, Map<String, String>> entry : getConfigKeys().entrySet()) {
```

Comment:
The existing logic is to read from `ServiceStore` first and uses the value in cConf as default. The logic didn't change.